clap_complete/src/dynamic/completer.rs

```diff
@@ -76,8 +76,61 @@ pub fn complete(
         } else if arg.is_escape() {
             is_escaped = true;
             state = ParseState::ValueDone;
-        } else if let Some(_long) = arg.to_long() {
-        } else if let Some(_short) = arg.to_short() {
+        } else if let Some((flag, value)) = arg.to_long() {
+            if let Ok(flag) = flag {
+                state = if let Some(opt) = current_cmd.get_arguments().find(|a| {
+                    a.get_long_and_visible_aliases()
+                        .map_or(false, |v| v.into_iter().find(|s| *s == flag).is_some())
+                }) {
+                    match opt.get_action() {
```

Comment:
what if we did
```rust
let opt = ...;
match opt.map(|o| o.get_action()) {
}
```
This would reduce some of the rightward drift and the trailing `else` that is hard to match up with the `if`

Reply:
This still needs resolving